Blinkenlights/blinkenlights.py

```diff
@@ -13,38 +13,108 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-# An exceedingly crude script to upload a bunch of png files to the matrix.
+# A script to upload a bunch of png files to the matrix.
 
+import argparse
+import contextlib
 import png
 import serial
 import sys
 
-if len(sys.argv) < 3 :
-    print("Usage: %s <gamma> <16x16 png file>..."%sys.argv[0], file=sys.stderr, );
-    exit(1)
-
-_InvGamma = 1/float(sys.argv[1])
-GammaLUT = [int(round(255*((v/255.0)**_InvGamma))) for v in range(256)]
-
-Serial = serial.Serial('/dev/ttyACM0', 9600)
-
-while True:
-    Serial.write("VER\n".encode())
-    l = Serial.readline()
-    print(l);
-    if l == b"ACK VER 1.0\r\n":
-        break
-Serial.write("ANM 600000\n".encode())
-print(Serial.readline())
-
-for png_file in sys.argv[2:]:
-    (width, height, rows, meta) = png.Reader(file=open(png_file, "rb")).asRGB8()
-    Serial.write("FRM 1000\n".encode())
-    print(Serial.readline())
-    for r in rows:
-        Serial.write(("RGB %s\n"%"".join("%02X"%GammaLUT[v] for v in r)).encode())
-        print(Serial.readline())
-Serial.write("DON\n".encode())
-print(Serial.readline())
-Serial.write("QUE\n".encode())
-print(Serial.readline())
+
+class BlinkenSerial(object):
+    def __init__(self, args):
+        self._serial = serial.Serial('/dev/ttyUSB0', 115200)
```

Comment:
It's fine for /dev/ttyUSB0 to be the default, but it may well be ttyUSB1, or this may run on Mac or Windows... we should make this a command-line flag.

Reply:
Done.